Refine the upgrade steps. (#4472)
istio/istio.io: content/docs/setup/kubernetes/upgrade/steps/index.md

`istio-system` namespace.

{{< warning >}}
Be sure to check out the [upgrade notice](/docs/setup/kubernetes/upgrade/notice)
for a concise list of things you should know before upgrading your deployment to Istio 1.2.
{{< /warning >}}

{{< tip >}}
Istio does **NOT** support skip level upgrades.  Only upgrades from 1.1 to 1.2
are supported. If you are on an older version, please upgrade to 1.1 first.
{{< /tip >}}

## Upgrade steps

[Download the new Istio release](/docs/setup/kubernetes/#downloading-the-release)
and change directory to the new release directory.

### Control plane upgrade

> The Istio control plane components include: Citadel, Ingress gateway, Egress gateway,

Reviewer: ```suggestion
```

> Pilot, Galley, Policy, Telemetry and Sidecar injector.  Choose one of the following

Reviewer 2: Do we want to list each control plane components?  This varies per profile.  The list here isn't correct for our default profile (should be without egress gateway but with Prometheus).

Reviewer: ```suggestion
```

Author: good point, hadn't considered this aspect of the document.  The upgrade process strongly depends on what the defaults are in `values.yaml` not the demo profile.

I'll deploy the default and add an info box (operators will need to turn those control plane components on which they had on prior).

> two **mutually exclusive** options to update the control plane:

Reviewer: ```suggestion
 Choose one of the following **mutually exclusive** options
 to update the control plane:
```

Author: done.

`istio-system` namespace.

{{< warning >}}
Be sure to check out the [upgrade notice](/docs/setup/kubernetes/upgrade/notice)
for a concise list of things you should know before upgrading your deployment to Istio 1.2.
{{< /warning >}}

{{< tip >}}

> Istio does **NOT** support skip level upgrades.  A skip level upgrade is an upgrade

Author: done

including configurations generated using
[helm template](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template).

> 1. Use `kubectl apply` to upgrade all the Istio's CRDs.  Wait a few seconds for the Kubernetes

Author: done.

      --namespace istio-system > $HOME/istio.yaml
    {{< /text >}}

>     If you want to enable [global mutual TLS](/docs/concepts/security/#mutual-tls-authentication),

Reviewer: @howardjohn is right. We should either link to the install instructions, or provide guidance for other helm options here too. @sdake Can you file a follow up issue for us to revisit this after release? Would that be ok with you @howardjohn?

>     If you want to enable [global mutual TLS](/docs/concepts/security/#mutual-tls-authentication),

Author: the only thing I changed was the alignment...

## Migrating per-service mutual TLS enablement via annotations to authentication policy

> If you use service annotations to override global mutual TLS enablement for a

Reviewer 3: @ security experts: is this still relevant today? It seemed to me this was a 1.0 -> 1.1 migration that was required, so it no longer needs to be documented. Is this the case?

Author: I'll leave this section to @yangminzhu to rework, and then rebase my PR on top.

Reviewer 2: Agreed, this should not be needed from 1.1 to 1.2.

  namespace: bar
  annotations:
    auth.istio.io/8000: NONE

> ---

Reviewer 3: Do we do this on our other samples?

Author: nah - we should for consistency - but I can revert this and others.

trafficPolicy:
  tls:
    mode: DISABLE
---
{{< /text >}}

> ## Migrating the `mtls_excluded_services` configuration to destination rules

Reviewer 3: Same comment as above, is this relevant for 1.1 -> 1.2?

Reviewer 2: I'm pretty sure all these security related stuff should be removed.

  trafficPolicy:
    tls:
      mode: DISABLE
---
{{< /text >}}

> ## Migrating from `RbacConfig` to `ClusterRbacConfig`

Reviewer 3: Same comment

Author: reverted.
